pkgs/top-level/all-packages.nix

```diff
@@ -30042,6 +30042,10 @@ with pkgs;
 
   sumokoin = callPackage ../applications/blockchains/sumokoin { boost = boost165; };
 
+  solana = callPackage ../applications/blockchains/solana {
```

Comment:
Should we call just the `solana` binary the “solana” package? I would expect “solana” to provide everything that the upstream installer does (`solana`, but also `solana-keygen`, `solana-validator`, `solana-test-validator`, `spl-token`, `cargo-build-bpf`, etc.)

But I suppose we can start with just the `solana` binary, and maybe even many of the others as well, and then add the remaining things once we figure out how to do that?

Reply:
You're right here. I was wondering also what best to call it. Perhaps solana-cli ?

Reply:
I called the package name "solana-testnet-cli". I had some hesitations about adding testnet in the name. I was wondering if perhaps we should also add the mainnet which is on a different version. I thought about starting with testnet so I could test it, then later on adding mainnet when I was confident that all is running as expected.

Reply:
> I was wondering if perhaps we should also add the mainnet which is on a different version. I thought about starting with testnet so I could test it, then later on adding mainnet when I was confident that all is running as expected.

Yeah this is a bit messy, because upstream releases just have a version number and sometimes they say “recommended for mainnet” and sometimes they say “this is a testnet release not reccommended for mainnet-beta”. It looks like they follow the release train model, where currently for example 1.8.x is recommended for mainnet, and 1.9.x for testnet, and at some point 1.9.y will become recommended for mainnet. So maybe we should offer the latest patch version of the latest testnet and mainnet release side by side? `solana18` and `solana19`, just like we have `python38` and `python39`? And then we can have a `solana` alias for the mainnet one.

Reply:
> You're right here. I was wondering also what best to call it. Perhaps solana-cli ?

After some thought, I think it’s fine to call it `solana`; if we add more binaries to the same package later that shouldn’t break users who were using the package for just the `solana` binary.

Reply:
how about having `solana-mainnet` and `solana-testnet` then ? It seems more transparent for end users. Otherwise they have to look up versions and what they mean.